common/namespace/namespace.go

```diff
@@ -253,6 +254,13 @@ func (ns *Namespace) IsOnCluster(clusterName string) bool {
 	return false
 }
 
+// FailoverHistory returns the a copy of failover history for this namespace.
+func (ns *Namespace) FailoverHistory() []*replication.FailoverStatus {
```

Comment:
Shall we return replication.FailoverStatus or persistence.FailoverStatus? cc @yux0 
Feel like I should return replication.FailoverStatus as that what we return when describing namespace. But this is a server side component, probably persistence.FailoverStatus also makes sense. 
Anyway I think a deep copy is needed here, similar to the ClusterName() method above.